### PR TITLE
Renaming SDKs

### PR DESCRIPTION
**Pull request by efgpinto**
References https://github.com/lightbend/kalix-proxy/issues/1941

I haven't renamed the folders yet.

### CODE REVIEW

**Comment by efgpinto** [comment]
As per @beritou suggestion, this PR changes the structure to this:

- Developing Services
  - Java SDK
  - Java/Scala Protobuf SDK
  - JavaScript/TypeScript Protobuf SDK
  - Run a service locally

_Originally posted by @beritou in https://github.com/lightbend/kalix-proxy/issues/1941#issuecomment-1442006408_

File: docs/src/modules/java/nav.adoc [hunk 1]
```diff
@@ -14,8 +13,7 @@
...
-*** xref:call-another-service.adoc[Calling other services]
-*** xref:failures-and-errors.adoc[Failures and Errors]
-*** xref:access-control.adoc[Access Control Lists (ACLs)]
-*** xref:java:api.adoc[Java API docs]
-*** xref:java:spring-client.adoc[Implementing Spring Client]
\ No newline at end of file
+*** xref:java:call-another-service.adoc[Calling other services]
+*** xref:java:failures-and-errors.adoc[Failures and Errors]
+*** xref:java:access-control.adoc[Access Control Lists (ACLs)]
+*** xref:java:using-jwts.adoc[Using JWTs in Spring]
```
Comment (franciscolopezsancho) [inline docs/src/modules/java/nav.adoc:19]:
```suggestion
*** xref:java:using-jwts.adoc[Using JWTs]
```

File: docs/src/modules/java/nav.adoc [hunk 2]
```diff
@@ -1,11 +1,10 @@
 ** xref:java:index.adoc[]
-*** xref:java:developer-tools.adoc[]
-*** xref:developing:development-process-java.adoc[]
-*** xref:java:quickstart-template.adoc[]
-*** xref:java:proto.adoc[]
+*** xref:java:getting-started.adoc[Getting Started]
+*** xref:java:spring-boot-integration.adoc[Spring Boot Integration]
+*** xref:java:development-process-spring.adoc[Development Process]
```
Comment (franciscolopezsancho) [inline docs/src/modules/java/nav.adoc:4]: maybe change the name of this file to `development-process.adoc`?

**Comment by franciscolopezsancho** [comment]
I just noticed that we have "getting started" in "Java SDK" but not in the other SDKs where we have "developer tools". Not only in this PR but in docs.kalix.io.
Shall we rename one or the other?

**Comment by franciscolopezsancho** [comment]
It looks good to me. As you can see, there are only tiny suggestions. I click as many links as I could until my trackpad ran out of battery. Couldn't find any of them broken or any misdirection.

File: docs/src/modules/java/nav.adoc (hunk 2 above)
Comment (efgpinto) [reply docs/src/modules/java/nav.adoc]: changed, tks

File: docs/src/modules/java/nav.adoc (hunk 1 above)
Comment (efgpinto) [reply docs/src/modules/java/nav.adoc]: changed, tks

**Comment by efgpinto** [comment]
> I just noticed that we have "getting started" in "Java SDK" but not in the other SDKs where we have "developer tools". Not only in this PR but in docs.kalix.io. Shall we rename one or the other?

They are actually slightly different pages in terms of structure and context though. Maybe the protocol-first ones should be converted into "Getting started".

**Comment by efgpinto** [comment]
Opening this up for final check. The bulk of the work was already reviewed in separate PRs anyway.

**Review by johanandren — APPROVED** [review]
LGTM (but was a "skim review")

**Pull request merged**